ping/ping_common.c

```diff
@@ -806,35 +813,51 @@ int gather_statistics(struct ping_rts *rts, uint8_t *icmph, int icmplen,
 		uint8_t *cp, *dp;
 
 		print_timestamp(rts);
-		printf(_("%d bytes from %s:"), cc, from);
+		if (!rts->opt_json)
+			printf(_("%d bytes from %s:"), cc, from);
+		construct_json(rts, PING_JSON_INT, "bytes", cc);
+		// TODO: split host IP address and name into separate attributes in JSON?
+		construct_json(rts, PING_JSON_STR, "host", from);
 
 		if (pr_reply)
-			pr_reply(icmph, cc);
+			pr_reply(rts, icmph, cc);
 
 		if (rts->opt_verbose && rts->ident != -1)
-			printf(_(" ident=%d"), ntohs(rts->ident));
+			if (!rts->opt_json)
+				printf(_(" ident=%d"), ntohs(rts->ident));
 
-		if (hops >= 0)
-			printf(_(" ttl=%d"), hops);
+		if (hops >= 0) {
+			if (!rts->opt_json)
+				printf(_(" ttl=%d"), hops);
```

Comment:
I also wonder if whole "json or plain output" shouldn't be somehow encapsulated into internal functions. This would be hard to use on `("%d bytes from %s:"), cc, from`, where two values are printed, but for simple printing e.g. `_(" ident=%d")` it could be doable:

```c
// ping/ping_json.h
#define PRINT(msg, json_msg, type, val) \
	do { \
		if (rts->opt_json) \
			printf(msg, val); \
		else \
			construct_json(rts, type, json_msg, val); \
	} while (0)
#define PRINT_INT(msg, json_msg, val) PRINT(msg, json_msg, val, PING_JSON_INT)
#define PRINT_STR(msg, json_msg, val) PRINT(msg, json_msg, val, PING_JSON_STR)

// here
PRINT_INT(_(" ident=%d"), "ident", hops);
```
Obviously it would help if we don't translate whole C parsing (e.g. `" ident=%d"` or `" ttl=%d"`), but just the word (e.g. `"ident"` or `"ttl"`). Also it's questionable why these technical words were translated.

Reply:
Good idea, thanks for the example, will check as well. But you are saying it would break translation of these terms if implementing it using this way, would that be acceptable or will I need to implement the additional gettext logic?

Reply:
The simple case (the above example which has duplicity of the strings) is ok. Additional gettext i18n improvements can be done later (unless you want to do it yourself). I don't want to overcomplicate the task.

Reply:
Added a commit implementing this, with some slight adjustments to allow for >= 1 format specifiers using a single macro. What do you think?

Reply:
LGTM, thank you!

Reply:
Great, thanks for checking.